### PR TITLE
Message comments and package-level README.md files

### PR DESCRIPTION
**Pull request by brawner**
This adds and cleans up some inline comments in the message and service files. I've also added README.md files which provide feature-level documentation and links to the message files.

New commit: https://github.com/ros2/rcl_interfaces/commit/9a6394a3eb69041936604fb7acd38038c56d79f4

### CODE REVIEW

**Comment by brawner** [comment]
@tfoote ping for review

File: action_msgs/msg/GoalStatus.msg [hunk 1]
```diff
@@ -1,15 +1,32 @@
-# An action goal can be in one of these states after it is accepted by an action server.
+# An action goal can be in one of these states after it is accepted by an action
+# server.
+#
 # For more information, see http://design.ros2.org/articles/actions.html
+
+# Indicates status has not been properly set
```
Comment (ahcorde) [inline action_msgs/msg/GoalStatus.msg]:
```suggestion
# Indicates status has not been properly set.
```

File: action_msgs/msg/GoalStatus.msg [hunk 2]
```diff
@@ -1,15 +1,32 @@
...
 int8 STATUS_CANCELED  = 5
+
+# The goal was terminated by the action server without an external request.
 int8 STATUS_ABORTED   = 6
 
 # Goal info (contains ID and timestamp)
 GoalInfo goal_info
 
-# Goal status
+# Action goal state-machine status
```
Comment (ahcorde) [inline action_msgs/msg/GoalStatus.msg]:
```suggestion
# Action goal state-machine status.
```

File: action_msgs/srv/CancelGoal.srv [hunk 3]
```diff
@@ -8,28 +8,39 @@
 # - If the goal ID is not zero and timestamp is not zero, cancel the goal with
 #   the given ID and all goals accepted at or before the timestamp.
 
-# Goal info containing an ID and timestamp
+# Goal info describing the goals to cancel, see above
```
Comment (ahcorde) [inline action_msgs/srv/CancelGoal.srv]:
```suggestion
# Goal info describing the goals to cancel, see above.
```

File: action_msgs/srv/CancelGoal.srv [hunk 4]
```diff
@@ -8,28 +8,39 @@
...
 GoalInfo goal_info
 ---
-# Return codes
+##
+## Return codes
```
Comment (ahcorde) [inline action_msgs/srv/CancelGoal.srv]:
```suggestion
## Return codes.
```

File: action_msgs/srv/CancelGoal.srv [hunk 5]
```diff
@@ -8,28 +8,39 @@
...
 # Indicates the goal is not cancelable because it is already in a terminal state.
-# No goals have transitioned to the CANCELING state.
-# The goals_canceling list is empty.
+#
+# No goals have transitioned to the CANCELING state. The goals_canceling list is
+# empty.
 int8 ERROR_GOAL_TERMINATED=3
 
-# Return code
+# Return code, see above definitions
```
Comment (ahcorde) [inline action_msgs/srv/CancelGoal.srv]:
```suggestion
# Return code, see above definitions.
```

File: action_msgs/srv/CancelGoal.srv [hunk 6]
```diff
@@ -8,28 +8,39 @@
...
 int8 return_code
+
 # Goals that accepted the cancel request
```
Comment (ahcorde) [inline action_msgs/srv/CancelGoal.srv]:
```suggestion
# Goals that accepted the cancel request.
```

File: builtin_interfaces/README.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+# builtin_interfaces
+A package containing message definitions for types defined in the OMG IDL Platform Specific Model.
+
+For more information about ROS 2 time, see [design.ros2.org](https://design.ros2.org/articles/clock_and_time.html)
```
Comment (ahcorde) [inline builtin_interfaces/README.md]:
```suggestion
For more information about ROS 2 time, see [design.ros2.org](https://design.ros2.org/articles/clock_and_time.html).
```

File: builtin_interfaces/README.md [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+
+For more information about ROS 2 interfaces, see [index.ros2.org](https://index.ros.org/doc/ros2/Concepts/About-ROS-Interfaces/)
```
Comment (ahcorde) [inline builtin_interfaces/README.md]:
```suggestion
For more information about ROS 2 interfaces, see [index.ros2.org](https://index.ros.org/doc/ros2/Concepts/About-ROS-Interfaces/).
```

File: builtin_interfaces/msg/Duration.msg [hunk 9]
```diff
@@ -1,2 +1,8 @@
+# Duration defines a period between two time points. It is comprised of a
+# seconds component and a nanoseconds component.
+
+# Seconds component, range is valid over any possible int32 value.
 int32 sec
+
+# Nanoseconds component in the range of [0, 10e9)
```
Comment (ahcorde) [inline builtin_interfaces/msg/Duration.msg]:
```suggestion
# Nanoseconds component in the range of [0, 10e9).
```

File: builtin_interfaces/msg/Time.msg [hunk 10]
```diff
@@ -1,2 +1,7 @@
+# Time indicates a specific point in time, relative to a clock's 0 point.
+
+# The seconds component, valid over all int32 values
```
Comment (ahcorde) [inline builtin_interfaces/msg/Time.msg]:
```suggestion
# The seconds component, valid over all int32 values.
```

File: builtin_interfaces/msg/Time.msg [hunk 11]
```diff
@@ -1,2 +1,7 @@
...
 int32 sec
+
+# The nanoseconds component, valid in the range [0, 10e9)
```
Comment (ahcorde) [inline builtin_interfaces/msg/Time.msg]:
```suggestion
# The nanoseconds component, valid in the range [0, 10e9).
```

File: composition_interfaces/README.md [hunk 12]
```diff
@@ -0,0 +1,9 @@
+# composition_interfaces
+This is a package containing message and service definitions for managing composable nodes in a container process. Generally these services are used by the ROS 2 [`roslaunch`](https://design.ros2.org/articles/roslaunch.html) system.
+
+For more information about ROS 2 interfaces, see [index.ros2.org](https://index.ros.org/doc/ros2/Concepts/About-ROS-Interfaces/)
```
Comment (ahcorde) [inline composition_interfaces/README.md]:
```suggestion
For more information about ROS 2 interfaces, see [index.ros2.org](https://index.ros.org/doc/ros2/Concepts/About-ROS-Interfaces/).
```

File: composition_interfaces/README.md [hunk 13]
```diff
@@ -0,0 +1,9 @@
...
+
+## Services (.srv)
+* [ListNodes](srv/ListNodes.srv): Retrieve a list of running composable nodes, including their names and ids.
+* [LoadNodes](srv/LoadNode.srv): Load a composable node
```
Comment (ahcorde) [inline composition_interfaces/README.md]:
```suggestion
* [LoadNodes](srv/LoadNode.srv): Load a composable node.
```

File: composition_interfaces/README.md [hunk 14]
```diff
@@ -0,0 +1,9 @@
...
+* [UnloadNode](UnloadNode.srv): Unload a specified node by its id
```
Comment (ahcorde) [inline composition_interfaces/README.md]:
```suggestion
* [UnloadNode](UnloadNode.srv): Unload a specified node by its id.
```

File: composition_interfaces/srv/LoadNode.srv [hunk 15]
```diff
@@ -1,28 +1,42 @@
-# A ROS package the composable node can be found in
+# The ROS package in which the composable node can be found
```
Comment (ahcorde) [inline composition_interfaces/srv/LoadNode.srv]:
```suggestion
# The ROS package in which the composable node can be found.
```

File: composition_interfaces/srv/LoadNode.srv [hunk 16]
```diff
@@ -1,28 +1,42 @@
...
 string package_name
-# a plugin within that package
+
+# A plugin within the ROS package "package_name"
```
Comment (ahcorde) [inline composition_interfaces/srv/LoadNode.srv]:
```suggestion
# A plugin within the ROS package "package_name".
```

File: composition_interfaces/srv/LoadNode.srv [hunk 17]
```diff
@@ -1,28 +1,42 @@
...
 string plugin_name
 
-# Name the composable node should use, or empty to use the node's default name
+# The assigned name of the composable node. Leave empty to use the node's
+# default name
```
Comment (ahcorde) [inline composition_interfaces/srv/LoadNode.srv]:
```suggestion
# default name.
```

File: composition_interfaces/srv/LoadNode.srv [hunk 18]
```diff
@@ -1,28 +1,42 @@
...
 string node_name
-# Namespace the composable node should use, or empty to use the node's default namespace
+
+# The assigned namespace of the composable node. Leave empty to use the node's
+# default namespace
```
Comment (ahcorde) [inline composition_interfaces/srv/LoadNode.srv]:
```suggestion
# default namespace.
```

File: composition_interfaces/srv/LoadNode.srv [hunk 19]
```diff
@@ -1,28 +1,42 @@
...
 string node_namespace
-# Values from message rcl_interfaces/Log
+
+# The assigned log level of the composable node. Enum values are found in
+# message rcl_interfaces/Log
```
Comment (ahcorde) [inline composition_interfaces/srv/LoadNode.srv]:
```suggestion
# message rcl_interfaces/Log.
```

File: composition_interfaces/srv/LoadNode.srv [hunk 20]
```diff
@@ -1,28 +1,42 @@
...
+
+# Remapping rules for this composable node.
+#
+# For more info about static_remapping rules and their syntax, see
+# https://design.ros2.org/articles/static_remapping.html
 # TODO(sloretz) rcl_interfaces message for remap rules?
 string[] remap_rules
-# Parameters to set
+
+# The Parameters of this composable node to set
```
Comment (ahcorde) [inline composition_interfaces/srv/LoadNode.srv]:
```suggestion
# The Parameters of this composable node to set.
```

File: composition_interfaces/srv/UnloadNode.srv [hunk 21]
```diff
@@ -3,5 +3,6 @@ uint64 unique_id
 ---
 # True if the node existed and was unloaded
 bool success
+
 # Human readable error message if success is false, else empty string
```
Comment (ahcorde) [inline composition_interfaces/srv/UnloadNode.srv]:
```suggestion
# Human readable error message if success is false, else empty string.
```

File: composition_interfaces/srv/UnloadNode.srv [hunk 22]
```diff
@@ -3,5 +3,6 @@ uint64 unique_id
 ---
 # True if the node existed and was unloaded
```
Comment (ahcorde) [inline composition_interfaces/srv/UnloadNode.srv]:
```suggestion
# True if the node existed and was unloaded.
```

File: composition_interfaces/srv/LoadNode.srv [hunk 23]
```diff
@@ -1,28 +1,42 @@
...
 # True if the node was successfully loaded
 bool success
+
 # Human readable error message if success is false, else empty string
 string error_message
+
 # Name of the loaded composable node (including namespace)
 string full_node_name
+
 # A unique identifier for the loaded node
```
Comment (ahcorde) [inline composition_interfaces/srv/LoadNode.srv]:
```suggestion
# A unique identifier for the loaded node.

File: composition_interfaces/srv/LoadNode.srv [hunk 24]
```diff
@@ -1,28 +1,42 @@
...
 string[] remap_rules
-# Parameters to set
+
+# The Parameters of this composable node to set
 rcl_interfaces/Parameter[] parameters
 
 # key/value arguments that are specific to a type of container process
 rcl_interfaces/Parameter[] extra_arguments
 ---
 # True if the node was successfully loaded
```
Comment (ahcorde) [inline composition_interfaces/srv/LoadNode.srv:32]:
```suggestion
# True if the node was successfully loaded.
```

File: composition_interfaces/srv/LoadNode.srv [hunk 25]
```diff
@@ -1,28 +1,42 @@
...
+# For more info about static_remapping rules and their syntax, see
+# https://design.ros2.org/articles/static_remapping.html
 # TODO(sloretz) rcl_interfaces message for remap rules?
 string[] remap_rules
-# Parameters to set
+
+# The Parameters of this composable node to set
 rcl_interfaces/Parameter[] parameters
 
 # key/value arguments that are specific to a type of container process
```
Comment (ahcorde) [inline composition_interfaces/srv/LoadNode.srv]:
```suggestion
# key/value arguments that are specific to a type of container process.
```

File: lifecycle_msgs/README.md [hunk 26]
```diff
@@ -0,0 +1,20 @@
+# lifecycle_msgs
+This package contains message and service definitions for managing lifecycle nodes.
+These messages and services form a standardized interface for transitioning these
+managed nodes through a known state-machine.
+
+For more information about life cycle nodes see: [design.ros2.org](http://design.ros2.org/articles/node_lifecycle.html)
```
Comment (ahcorde) [inline lifecycle_msgs/README.md]:
```suggestion
For more information about life cycle nodes see: [design.ros2.org](http://design.ros2.org/articles/node_lifecycle.html).
```

File: lifecycle_msgs/README.md [hunk 27]
```diff
@@ -0,0 +1,20 @@
...
+
+For more information about ROS 2 interfaces, see [index.ros2.org](https://index.ros.org/doc/ros2/Concepts/About-ROS-Interfaces/)
```
Comment (ahcorde) [inline lifecycle_msgs/README.md]:
```suggestion
For more information about ROS 2 interfaces, see [index.ros2.org](https://index.ros.org/doc/ros2/Concepts/About-ROS-Interfaces/).
```

File: lifecycle_msgs/README.md [hunk 28]
```diff
@@ -0,0 +1,20 @@
...
+
+## Messages (.msg)
+* [State](msg/State.msg): A lifecycle node's state-machine state
+* [Transition](msg/Transition.msg): A state transition with specific id and label
+* [TransitionDescription](msg/TransitionDescription.msg): A description of a transition from one state-machine state to another.
+* [TransitionEvent](msg/TransitionEvent.msg): A timestamped state transition
```
Comment (ahcorde) [inline lifecycle_msgs/README.md]:
```suggestion
* [State](msg/State.msg): A lifecycle node's state-machine state.
* [Transition](msg/Transition.msg): A state transition with specific id and label.
* [TransitionDescription](msg/TransitionDescription.msg): A description of a transition from one state-machine state to another.
* [TransitionEvent](msg/TransitionEvent.msg): A timestamped state transition.
```

File: lifecycle_msgs/README.md [hunk 29]
```diff
@@ -0,0 +1,20 @@
...
+
+## Services (.srv)
+* [ChangeState](srv/ChangeState.srv): Request a node change states with a specific transition
+* [GetAvailableStates](srv/GetAvailableStates.srv): Request an array of states that this node can transition to.
+* [GetAvailableTransitions](srv/GetAvailableTransitions.srv): Request an array of lifecycle state transitions available for this node
+* [GetState](srv/GetState.srv): Request the current lifecycle state of this node
```
Comment (ahcorde) [inline lifecycle_msgs/README.md]:
```suggestion
* [ChangeState](srv/ChangeState.srv): Request a node change states with a specific transition.
* [GetAvailableStates](srv/GetAvailableStates.srv): Request an array of states that this node can transition to.
* [GetAvailableTransitions](srv/GetAvailableTransitions.srv): Request an array of lifecycle state transitions available for this node.
* [GetState](srv/GetState.srv): Request the current lifecycle state of this node.
```

File: rosgraph_msgs/README.md [hunk 30]
```diff
@@ -0,0 +1,9 @@
+# rosgraph_msgs
+This is a package containing message definitions relating to the ROS Computation Graph. These are generally considered to be low-level messages that end users do not interact with.
+
+For more information about ROS 2 clock, see [design.ros2.org](https://design.ros2.org/articles/clock_and_time.html)
+
+For more information about ROS 2 interfaces, see [index.ros2.org](https://index.ros.org/doc/ros2/Concepts/About-ROS-Interfaces/)
```
Comment (ahcorde) [inline rosgraph_msgs/README.md]:
```suggestion
For more information about ROS 2 interfaces, see [index.ros2.org](https://index.ros.org/doc/ros2/Concepts/About-ROS-Interfaces/).
```

File: rcl_interfaces/srv/SetParametersAtomically.srv [hunk 31]
```diff
@@ -1,7 +1,8 @@
 # A list of parameters to set atomically.
+#
 # This call will either set all values, or none of the values.
 Parameter[] parameters
 
 ---
-
+# Indicates whether setting all of the parameters succeeded or not and why
```
Comment (ahcorde) [inline rcl_interfaces/srv/SetParametersAtomically.srv]:
```suggestion
# Indicates whether setting all of the parameters succeeded or not and why.
```

File: rcl_interfaces/srv/SetParameters.srv [hunk 32]
```diff
@@ -2,5 +2,5 @@
 Parameter[] parameters
 
 ---
-
+# Indicates whether setting each parameter succeeded or not and why
```
Comment (ahcorde) [inline rcl_interfaces/srv/SetParameters.srv]:
```suggestion
# Indicates whether setting each parameter succeeded or not and why.
```

File: rcl_interfaces/srv/ListParameters.srv [hunk 33]
```diff
@@ -1,11 +1,14 @@
...
 string[] prefixes
+
 # Relative depth from given prefixes to return.
+#
 # Use DEPTH_RECURSIVE to get the recursive parameters and prefixes for each prefix.
 uint64 depth
 
 ---
-
+# The list of parameter names and their prefixes
```
Comment (ahcorde) [inline rcl_interfaces/srv/ListParameters.srv]:
```suggestion
# The list of parameter names and their prefixes.
```

File: rcl_interfaces/srv/ListParameters.srv [hunk 34]
```diff
@@ -1,11 +1,14 @@
+# Recursively get parameters with unlimited depth
```
Comment (ahcorde) [inline rcl_interfaces/srv/ListParameters.srv]:
```suggestion
# Recursively get parameters with unlimited depth.
```

File: rcl_interfaces/msg/ParameterValue.msg [hunk 35]
```diff
@@ -1,17 +1,38 @@
 # Used to determine which of the next *_value fields are set.
 # ParameterType.PARAMETER_NOT_SET indicates that the parameter was not set
 # (if gotten) or is uninitialized.
-# Values are enumerated in ParameterType.msg
+# Values are enumerated in `ParameterType.msg`
```
Comment (ahcorde) [inline rcl_interfaces/msg/ParameterValue.msg]: Punctuation

File: rcl_interfaces/msg/ParameterEvent.msg [hunk 36]
```diff
@@ -1,12 +1,18 @@
 # This message contains a parameter event.
-# It was an atomic update.
-# A specific parameter name can only be in one of the three sets.
+# Because the parameter event was an atomic update, a specific parameter name
+# can only be in one of the three sets.
 
+# The time stamp when this parameter event occurred
```
Comment (ahcorde) [inline rcl_interfaces/msg/ParameterEvent.msg]: Punctuation

**Review by ahcorde — CHANGES_REQUESTED** [review]
Unified the style of the comments with punctuation

File: composition_interfaces/srv/LoadNode.srv (hunk 24 above)
Comment (brawner) [reply composition_interfaces/srv/LoadNode.srv:32]: Added periods here and through out most other messages.

**Comment by brawner** [comment]
Addressed punctuation comments and rebased on master. This PR no longer depends on others.

**Comment by brawner** [comment]
Closing as this is also captured in #92.

**Pull request closed without merging**